File: Go/quicksort.go
package main

import (
	"math/rand"
)

func quicksort(vec []int) []int {
	if len(vec) <= 1 {
		return vec
	}

	middlepos := len(vec) / 2
	middle := vec[middlepos]

	var lesser []int
	var greater []int

	newvec := append(vec[:middlepos], vec[middlepos+1:]...)

	for _, el := range newvec {
		if el <= middle {
			lesser = append(lesser, el)
			continue
		}
		greater = append(greater, el)
	}

	var out []int
	lvals := quicksort(lesser)
	gvals := quicksort(greater)
	out = append(out, lvals...)
	out = append(out, middle)
	out = append(out, gvals...)
	return out
}

func main() {
	var vec = []int{}

	for x := 0; x < 1000000; x++ {
		vec = append(vec, rand.Intn(1000000))
	}

	quicksort(vec)
}

File: Python/quicksort.py
import random
import sys

sys.setrecursionlimit(20000000)

def quicksort(vec):
    if len(vec) <= 1:
        return vec

    middlepos = len(vec) / 2
    middle = vec[middlepos]

    lesser = []
    greater = []

    vec.pop(middlepos)

    for el in vec:
        if el <= middle:
            lesser.append(el)
            continue
        greater.append(el)

    return quicksort(lesser)+[middle]+quicksort(greater)

if __name__ == '__main__':
    vec = []
    for x in range(1000000):
        vec.append(random.randint(0,x))
    
    quicksort(vec)
